### PR TITLE
Added support to build and run UI only apps.

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Aio commands used to error out when UI only apps were run.
Made sure all action dependencies for their build and deploy only are called if manifest file is present.

## How Has This Been Tested?

create UI only app from CLI. Try out aio app run command.
## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/build.actions.js [hunk 1]
```diff
@@ -33,95 +33,97 @@ class BuildActions extends BaseScript {
    * @memberof DeployActions
    */
   async run (args = [], buildConfig = {}) {
-    const taskName = 'Build actions'
-    this.emit('start', taskName)
+    if (this.config.app.hasBackend) {
```
Comment (moritzraho) [inline scripts/build.actions.js]: to be consistent here I would do like https://github.com/adobe/aio-app-scripts/blob/master/scripts/deploy.ui.js#L29

File: scripts/deploy.actions.js [hunk 2]
```diff
@@ -39,52 +39,54 @@ class DeployActions extends BaseScript {
    * @memberof DeployActions
    */
   async run (args = [], deployConfig = {}) {
-    const taskName = 'Deploy actions'
-    this.emit('start', taskName)
+    if (this.config.app.hasBackend) {
```
Comment (moritzraho) [inline scripts/deploy.actions.js]: same comment here do like https://github.com/adobe/aio-app-scripts/blob/master/scripts/deploy.ui.js#L29

File: scripts/dev.js [hunk 3]
```diff
@@ -127,10 +127,10 @@ class ActionServer extends BaseScript {
       if (hasFrontend) {
```
Comment (moritzraho) [inline scripts/dev.js]: also if not hasBackend we don't want to build & deploy actions + don't need to start the openwhisk server

**Comment by purplecabbage** [comment]
this has failing tests, and now merge conflicts also.

**Comment by sandeep-paliwal** [comment]
@purplecabbage I will update the PR

File: test/scripts/build.ui.test.js [hunk 4]
```diff
@@ -112,3 +112,13 @@ test('should generate and inject web and non web action urls into web-src/src/co
...
 })
+
+test('No backend is present', async () => {
+  global.loadFs(vol, 'sample-app')
+  mockAIOConfig.get.mockReturnValue(global.fakeConfig.tvm)
+  vol.unlinkSync('./manifest.yml')
+
+  const scripts = await AppScripts()
+  await scripts.buildUI()
+  expect(vol.existsSync('/web-src/src/config.json')).toBe(true)
```
Comment (moritzraho) [inline test/scripts/build.ui.test.js]: would be good if you add some expect statements like config.json === {} and scripts._config.app.hasBackend ===false

File: test/scripts/dev.test.js [hunk 5]
```diff
@@ -920,3 +920,13 @@ describe('port unavailable', () => {
...
 })
+
+test('No backend is present', async () => {
+  global.loadFs(vol, 'sample-app')
+  mockAIOConfig.get.mockReturnValue(global.fakeConfig.tvm)
+  vol.unlinkSync('./manifest.yml')
+
+  const scripts = await AppScripts()
+  await scripts.runDev()
+  expect(vol.existsSync('/web-src/src/config.json')).toEqual(true)
```
Comment (moritzraho) [inline test/scripts/dev.test.js]: Here I would put this test as part of a describe (ui only) and add a few more tests:
- ui server has started
- no ow server started
- no build & deploy called
- only ui config in vscode file
- config.json = {}
- scripts._config.hasBackend = false

**Pull request merged**